core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -218,7 +218,13 @@ private[spark] class DiskBlockObjectWriter(
     recordWritten()
   }
 
-  override def write(b: Int): Unit = throw new UnsupportedOperationException()
+  override def write(b: Int): Unit = {
```

Comment:
I think that we used to call this via a DataOutputStream, but that step is no longer necessary. I'll roll this back.